### DISCUSSION TITLE
Embed sha in assembly?

### DISCUSSION THREAD

**Discussion by sgrassie** [adamralph/minver#758] [Q&A] [answered]
Hi, firstly can I say that coming from GitVersion, that this lib is _much_ simpler and is removing some complications for me as part of the build pipeline.

Second, I might be missing something so I wanted to ask this in a discussion before I started an issue, but I can't see a way of getting the `sha` into the `<InformationalVersion>`. Looking at the code, the `sha` is clearly used as part of the Versioner.cs#L120, so it's available. Am I missing someway of getting it into the `buildMetaData`?

Some people just want to see the `sha` as part of the extended version information 😄

**Comment by sgrassie** [comment]
Ok, I can see how I could do this in an `AfterTargets="MinVer"` by pulling the required data out of the CI environment. It would still be nice to be able to do this locally though.

**Reply by bording** [reply]
@sgrassie You should consider adding a package reference to `Microsoft.SourceLink.GitHub` to your project. Part of what it does is add the commit hash to the informational version.

I believe this is one of the reasons that MinVer doesn't do this. If it did, then you'd end up with the hash in the version twice.

**Reply by sgrassie** [reply]
Well of course that makes perfect sense. Thanks. 

Except if the client you are working with uses a non-standard VCS that you've adapted MinVer to work against... Not sure the effort of SourceLink integration as well is going to be worth it. 

Thanks again.

**Reply by bording** [reply]
There are Source Link packages available for many different source control providers. Take a look at https://github.com/dotnet/sourcelink to see a list. Maybe there is one that works for you already.

**Reply by adamralph** [reply]
@bording would you consider re-posting your suggestion to use Source Link as a top level _answer_? Then we can mark it as the answer.

**Reply by bording** [reply]
It seems very odd that a reply can't be marked as an answer!

**Comment by bording** [comment, marked as answer]
You should consider adding a package reference to Microsoft.SourceLink.GitHub to your project. Part of what it does is add the commit hash to the informational version.

I believe this is one of the reasons that MinVer doesn't do this. If it did, then you'd end up with the hash in the version twice.

If your code isn't on GitHub, there are Source Link packages available for many different source control providers. Take a look at https://github.com/dotnet/sourcelink to see a list. Maybe there is one that works for you already.

**Reply by adamralph** [reply]
Update—the .NET 8 SDK provides this as [`SourceRevisionId`](https://learn.microsoft.com/en-us/dotnet/core/compatibility/sdk/8.0/source-link#new-behavior).
